packages/flutter_tools/lib/src/windows/visual_studio_solution_utils.dart

```diff
@@ -0,0 +1,318 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../convert.dart';
+import '../globals.dart' as globals;
+import '../plugins.dart';
+import '../project.dart';
+import 'visual_studio_project.dart';
+
+// Constants corresponding to specific reference types in a solution file.
+// These values are defined by the .sln format.
+const String _solutionTypeGuidFolder = '2150E333-8FDC-42A3-9474-1A3956D46DE8';
+const String _solutionTypeGuidVcxproj = '8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942';
+
+// The GUID for the folder above, managed by this class. This is an arbitrary
+// value that was randomly generated, but it should not be changed since that
+// would cause issues for existing Flutter projects.
+const String _flutterPluginSolutionFolderGuid = '5C2E738A-1DD3-445A-AAC8-EEB9648DD07C';
+// The FlutterBuild project GUID. This is an arbitrary
+// value that was randomly generated, but it should not be changed since that
+// would cause issues for existing Flutter projects.
+const String _flutterBuildProjectGuid = '6419BF13-6ECD-4CD2-9E85-E566A1F03F8F';
+
+/// Extracts and stores the plugin name and vcxproj GUID for [plugin].
+class _PluginProjectInfo {
+  _PluginProjectInfo(Plugin plugin) {
+    name = plugin.name;
+    final File projectFile = globals.fs.directory(plugin.path).childDirectory('windows').childFile('plugin.vcxproj');
+    guid = VisualStudioProject(projectFile).guid;
+    if (guid == null) {
+      throwToolExit('Unable to find a plugin.vcxproj for plugin "$name"');
+    }
+  }
+
+  // The name of the plugin, which is also the name of the symlink folder.
+  String name;
+
+  // The GUID of the plugin's project.
+  String guid;
+}
+
+/// Updates the solution file for [project] to have the project references and
+/// dependencies to include [plugins], removing any previous plugins from the
+/// solution.
+Future<void> updatePluginsInSolution(WindowsProject project, List<Plugin> plugins) async {
```

Comment:
Consider refactoring this series of methods into a class

Reply:
I wrapped it in a class, added a filesystem injection, and moved the project to the constructor. Not the most elegant class, but at least it has the testability property of not using global. We can revisit if/when we do a full parser.